### PR TITLE
Deployment with Terraform on AWS

### PR DESCRIPTION
**Pull request by daftano**

### CODE REVIEW

File: deploy/ecs.tf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+resource "aws_ecs_cluster" "fargate" {
```
Comment (rjw57) [inline deploy/ecs.tf]: This file needs a comment at the top explaining what resources it defines and how they are linked.

File: deploy/iam.tf [hunk 2]
```diff
@@ -0,0 +1,47 @@
+resource "aws_iam_role" "ecs_task_assume" {
```
Comment (rjw57) [inline deploy/iam.tf]: This file needs a comment at the top explaining what resources it defines and how they are linked.

File: deploy/versions.tf [hunk 3]
```diff
@@ -0,0 +1,4 @@
+
+terraform {
+  required_version = ">= 0.12"
```
Comment (rjw57) [inline deploy/versions.tf]: This is a very useful annotation.

File: docs/Terraform.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+You can deploy the application and its dependencies with Terraform.
```
Comment (rjw57) [inline docs/Terraform.md]: This commit should be folded into the previous one(!)

Comment (rjw57) [inline docs/Terraform.md]: This document lacks a title.

File: deploy/vars.tf [hunk 5]
```diff
@@ -0,0 +1,60 @@
+terraform {
```
Comment (rjw57) [inline deploy/vars.tf]: The idiomatic name for this is `variables.tf`.

File: deploy/sg_ecs.tf [hunk 6]
```diff
@@ -0,0 +1,46 @@
+resource "aws_security_group" "ecs" {
+  name        = "ecs"
+  description = "Allow traffic for ecs"
+  vpc_id      = module.base_vpc.vpc_id
+
+  ingress {
+    from_port   = 0
+    to_port     = 65535
```
Comment (sdwenham) [inline deploy/sg_ecs.tf]: This scope is to wide as we know the ports needed

File: deploy/sg_ecs.tf [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    protocol    = "tcp"
+    cidr_blocks = split(",", var.CIDR_PUBLIC)
+  }
+
+  ingress {
+    from_port   = 0
+    to_port     = 65535
```
Comment (sdwenham) [inline deploy/sg_ecs.tf]: This scope is to wide as we know the ports needed

File: deploy/sg_ecs.tf [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+resource "aws_security_group" "alb" {
+  name        = "alb"
+  description = "Allow traffic for alb"
+  vpc_id      = module.base_vpc.vpc_id
+
+  ingress {
+    from_port   = 80
+    to_port     = 8000
```
Comment (sdwenham) [inline deploy/sg_ecs.tf]: This scope is to wide as we know the ports needed and this missed port 8025 meaning the mailhog instance cant be accessed

**Pull request closed without merging**